helpers/compose.go

```diff
@@ -0,0 +1,166 @@
+	return client.New(ctx)
+}
+
+func GetVersion(ctx context.Context, c *client.Client) (string, error) {
+	versionResult, err := mobycli.ExecSilent(ctx)
+	if versionResult == nil {
+		return "", err
+	}
+	return string(versionResult), err
+}
+
+func Up(ctx context.Context, c client.Client, projectName, composeFilePath string, detach bool) error {
+	if projectName == "" {
+		projectName = "meshery"
+	}
+	project := types.Project{Name: projectName, WorkingDir: filepath.Dir(composeFilePath), ComposeFiles: []string{composeFilePath}}
+
+	var logConsumer api.LogConsumer
+	if detach {
+		_, pipeWriter := io.Pipe()
+		logConsumer = format.NewLogConsumer(ctx, pipeWriter, false, false)
+	}
+	return c.ComposeService().Up(ctx, &project, api.UpOptions{Start: api.StartOptions{Attach: logConsumer}})
+}
+
+func Rm(ctx context.Context, c client.Client, projectName, composeFilePath string, force bool) error {
+	if projectName == "" {
+		projectName = "meshery"
+	}
+	project := types.Project{Name: projectName, WorkingDir: filepath.Dir(composeFilePath), ComposeFiles: []string{composeFilePath}}
+
+	return c.ComposeService().Remove(ctx, &project, api.RemoveOptions{Force: force})
+}
+
+func Stop(ctx context.Context, c client.Client, projectName, composeFilePath string, force bool) error {
+	if projectName == "" {
+		projectName = "meshery"
+	}
+	project := types.Project{Name: projectName, WorkingDir: filepath.Dir(composeFilePath), ComposeFiles: []string{composeFilePath}}
+
+	return c.ComposeService().Stop(ctx, &project, api.StopOptions{})
+}
+
+func Ps(ctx context.Context, c *client.Client, all, quiet bool, formatOpt string) error {
+	containerList, err := c.ContainerService().List(ctx, all)
+	if err != nil {
+		return errors.Wrap(err, "fetch containers")
+	}
+
+	if quiet {
+		for _, c := range containerList {
+			fmt.Println(c.ID)
+		}
+		return nil
+	}
+
+	view := getViewFromContainerList(containerList)
+	return format.Print(view, formatOpt, os.Stdout, func(w io.Writer) {
+		for _, c := range view {
+			_, _ = fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\n", c.ID, c.Image, c.Command, c.Status,
+				strings.Join(c.Ports, ", "))
+		}
+	}, "CONTAINER ID", "IMAGE", "COMMAND", "STATUS", "PORTS")
+}
+
+func ListContainers(ctx context.Context, c *client.Client, all bool) ([]ContainerView, error) {
+	containerList, err := c.ContainerService().List(ctx, all)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch containers: %w", err)
+	}
+
+	view := getViewFromContainerList(containerList)
+	return view, nil
+}
+
+func getViewFromContainerList(containerList []containers.Container) []ContainerView {
+	retList := make([]ContainerView, len(containerList))
+	for i, c := range containerList {
+		retList[i] = ContainerView{
+			ID:      c.ID,
+			Image:   c.Image,
+			Status:  c.Status,
+			Command: c.Command,
+			Ports:   formatter.PortsToStrings(c.Ports, getFQDN(c)),
+		}
+	}
+	return retList
+}
+func getFQDN(container containers.Container) string {
```

Comment:
Is there a specific need for this function?
How did we decide on which docker compose command to support/not support?
Is there some prior discussion, pls point me to it.

Reply:
> Is there a specific need for this function?

There isn't a specific need per se, I've tried to keep the implementation parallel to the [compose-cli](https://github.com/docker/compose-cli/blob/main/cli/cmd/ps.go)

> How did we decide on which docker compose command to support/not support?

I've analysed the usage in meshery to see which commands are needed, there's a PR there in which these functions will be used, replacing the `exec`s.

> Is there some prior discussion, pls point me to it.

There isn't. If you know of any compose commands used by any meshery component do let me know and I'll add those too.

Reply:
So far I have seen instances of the below docker-compose commands in Meshery, so this should be fine. We can add more if required.

- up -d
- logs -f
- ps
- stop
- rm -f
- pull 

This LGTM.